lib/gtfs/components/gtfs-search.js

```diff
@@ -0,0 +1,255 @@
+    return (
+      <span style={{ color: 'black' }}>
+        {option.stop
+          ? <Glyphicon glyph='map-marker' />
+          : <Glyphicon glyph='option-horizontal' />
+        }
+        {' '}{option.label}{' '}
+        <Label>
+          {option.agency && option.agency.name ? option.agency.name : ''}
+        </Label>
+      </span>
+    )
+  }
+
+  _sortOptions = (a: Option, b: Option, input: string) => {
+    const aLabel = a.label.toLowerCase()
+    const bLabel = b.label.toLowerCase()
+    if (aLabel.startsWith(input)) {
+      return bLabel.startsWith(input) ? aLabel.localeCompare(bLabel) : -1
+    } else {
+      return bLabel.startsWith(input) ? 1 : aLabel.localeCompare(bLabel)
+    }
+  }
+
+  _entityToOption = (entity: GtfsStop | GtfsRoute, agency: Feed) => {
+    if (entity.stop_id) {
+      const stop: GtfsStop = ((entity: any): GtfsStop)
+      const stopCode = stop.stop_code ? stop.stop_code : stop.stop_id
+      return {
+        stop,
+        label: `${stop.stop_name} (${stopCode})`,
+        value: stop.stop_id,
+        agency
+      }
+    } else {
+      const route: GtfsRoute = ((entity: any): GtfsRoute)
+      return {
+        route,
+        label: `${this.getRouteName(route)} (${route.route_id})`,
+        value: route.route_id,
+        agency
+      }
+    }
+  }
+
+  getRouteName = (route: GtfsRoute): string => {
+    const {route_short_name: shortName, route_long_name: longName} = route
+    if (shortName && longName) return `${shortName} - ${longName}`
+    else if (longName) return longName
+    else return shortName || '[no name]'
+  }
+
+  /**
+   * Request stop and route entities from GraphQL endpoint.
+   * @param  {String} textInput search string
+   * @param  {Object} feed      feed source to search
+   * @param  {[type]} entities  entity types to search (stops and/or routes)
+   * @param  {[type]} filterByRoute route on which to filter stops
+   * @param  {[type]} filterByStop  stop on which to filter routes
+   */
+  _searchEntitiesWithString = (textInput: string, feed: Feed, entities: Array<string>, filterByRoute: GtfsRoute, filterByStop: GtfsStop) => {
+    const namespace = feed.publishedVersionId
+    const {user} = this.props
+    // FIXME: Handle filter by route/stop (filter by route will query routes
+    // with a specified route id and stops nested underneath)!!!
+    const variables = {
+      routeId: filterByRoute ? filterByRoute.route_id : undefined,
+      stopId: filterByStop ? filterByStop.stop_id : undefined,
+      namespace,
+      search: textInput
+    }
+    const queryForRoutes = entities.indexOf('routes') !== -1
+    const queryForStops = entities.indexOf('stops') !== -1
+    const includeRouteSubQuery = !!filterByStop
+    const includeStopSubQuery = !!filterByRoute
+    // Build routes/stops query based on entities searching and whether we're
+    // filtering by a specific route or stop ID.
+    const body = JSON.stringify({
+      query: `
```

Comment:
Let's move this graphql to the graphql file.

Reply:
This is a pretty complex construction of a GraphQL query.  I don't know how much sense it would make to have it live outside of this context, not to mention we would be passing like six arguments to the function.